### PR TITLE
Menu and tagging

### PR DESCRIPTION
**Pull request by pbywater**
- Added shuffle and search buttons
- Animation on tag/search menu
- Added unique tags based on data
- Added ability to filter by tag by clicking on an individual tag or by removing tags
- Added ability to clear filter
#14

### CODE REVIEW

**Comment by joeylouise** [comment]
This PR also
-adds the sticky menu to the homepage
-adds all icons into assets folder

Relates #15 #7

**Comment by ghost** [comment]
Also links up the main three 'source' nodes to one another
relates #52

**Review by des-des — CHANGES_REQUESTED** [review]
Remove all the DS_Store files, change your git ignore so these do not come up again..
`public/assets/icons/menu/notification/_DS_Store`

File: .gitignore [hunk 1]
```diff
@@ -59,6 +59,11 @@ typings/
 
 config.env
 
+# DS store
+.DS_Store
```
Comment (des-des) [inline .gitignore]: **/*DS_Store

might do it ..?

File: public/main.css [hunk 2]
```diff
@@ -34,6 +38,393 @@
...
+}
+
+.menu {
+  position: absolute;
+  height: 15%;
+  box-sizing: border-box;
+  text-align: center;
+  width: 100%;
+  bottom: 0;
+  z-index: 999999;
```
Comment (des-des) [inline public/main.css]: `z-index: 2` is prob okay

File: public/main.css [hunk 3]
```diff
@@ -34,6 +38,393 @@
...
+    display: flex;
+    justify-content: space-around;
+    cursor: pointer;
+    top: 30%;
+ }
+
+ .tag-container {
+   text-align: center;
+   height: 50%;
+     display: none;
```
Comment (des-des) [inline public/main.css]: indentaion

File: public/main.css [hunk 4]
```diff
@@ -34,6 +38,393 @@
...
+     justify-content: space-around;
+   list-style-type: none;
+   background: rgb(142,165,206);
+   border-radius: 30%;
+   align-items: center;
+ }
+
+
+ .tags li {
+      list-style-type: none;
```
Comment (des-des) [inline public/main.css]: indentation

File: public/main.css [hunk 5]
```diff
@@ -34,6 +38,393 @@
...
+      display: none;
+      text-align: center;
+      height: 50%;
+ }
+
+.top-bar {
+  position: absolute;
+  z-index: 99999;
```
Comment (des-des) [inline public/main.css]: too big

File: public/main.css [hunk 6]
```diff
@@ -34,6 +38,393 @@
...
+#to-extend {
+  transform-origin: bottom;
+  transition: transform 0.6s ease;
+}
+
+
+.underline {
+  width:90%;
+  height: 90%;
+    display: inline-block;
```
Comment (des-des) [inline public/main.css]: indentation

File: public/main.css [hunk 7]
```diff
@@ -34,6 +38,393 @@
...
+    position: absolute;
+    z-index: -1;
+  right: 5%;
+
+}
+
+.underline:after {
```
Comment (des-des) [inline public/main.css]: is underline the best class name?

File: public/main.css [hunk 8]
```diff
@@ -34,6 +38,393 @@
...
+  text-align: right;
+    background: transparent !important;
+}
+
+.close-icon {
+  width: 40%;
+}
+
+.filter-tags {
+width: 10%;
```
Comment (des-des) [inline public/main.css]: indentaion

File: public/main.css [hunk 9]
```diff
@@ -34,6 +38,393 @@
...
+}
+
+.underline.active:after {
+    width: 100%;
+    background: white;
+}
+
+.close-tags {
+  text-align: right;
+    background: transparent !important;
```
Comment (des-des) [inline public/main.css]: indentation

File: public/scripts/main.js [hunk 10]
```diff
@@ -0,0 +1,84 @@
+document.addEventListener('DOMContentLoaded', (event) => {
```
Comment (des-des) [inline public/scripts/main.js:1]: This file needs to be refactored, heavily. Talk to me if you do not know how

File: src/client/d3/d3.js [hunk 11]
```diff
@@ -15,6 +16,20 @@ d3.json(url, (err, data) => {
   });
   // taggedNodesByTag returns an object with the cx and cy for the central node within each tag group
   const taggedNodesByTag = tagNodesByTag(sortedWithMax, 160, 120, generateId());
+  // Add unique tags to tag list for user to select from
+  Object.keys(taggedNodesByTag).forEach((tag) => {
+    $('.tags').append(
+      `<li class='tag-container ${tag}'>
```
Comment (des-des) [inline src/client/d3/d3.js]: Vulnerable to script injection attacks

File: src/client/helpers/helpers.js [hunk 12]
```diff
@@ -10,6 +10,100 @@ const binByKey = (key, xs) =>
...
+    $(this).parent().hide();
+    showMemoriesByActiveTags();
+  });
+}
+
+function closeTagMenu() {
+  $('.close-tags').on('click', () => {
+    $('.underline').removeClass('active');
+    setTimeout(() => {
+      $('.shuffle-memories').fadeIn(1000);
```
Comment (des-des) [inline src/client/helpers/helpers.js]: You have variables like, 1000, and 500 scattered around everywhere.

Make a variable?

**Review by des-des — CHANGES_REQUESTED** [review]

File: public/main.css (hunk 7 above)
Comment (pbywater) [reply public/main.css]: It refers to a line that extends underneath the tags - would there be a better name for it?

File: public/main.css (hunk 2 above)
Comment (pbywater) [reply public/main.css]: Haha I'll change these

File: public/scripts/main.js (hunk 10 above)
Comment (joeylouise) [reply public/scripts/main.js:1]: Nice and refactored now! 👍

**Review by des-des — APPROVED** [review]

**Pull request merged**